### PR TITLE
[CAZB-2142] Account set up page

### PR DESCRIPTION
**Pull request by pkalamarski**
## Motivation and Context

https://eaflood.atlassian.net/browse/CAZB-2142

## Description

Created view for setting up an account of invited user, added validation, communication with api, error handling and tests.

## How Has This Been Tested?

Manually, rspec and cucumber tests.
A valid token has to be provided as a query parameter, token is available in Account API logs after adding the new user.

## Screenshots (if appropriate):
image
image
image

## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [x] Includes link to an issue (if apply)
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: app/forms/set_up_account_form.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+##
+# This class is used to validate user data filled in +app/views/users/set_up.html.haml+.
+class SetUpAccountForm < NewUserBaseForm
```
Comment (likeuwill) [inline app/forms/set_up_account_form.rb:5]: `... < BaseForm`

File: app/views/users_management/users/set_up.html.haml [hunk 2]
```diff
@@ -1,8 +1,62 @@
...
+            There is a problem
+          .govuk-error-summary__body
+            %ul.govuk-list.govuk-error-summary__list
+              %li
+                - @errors.uniq(&:second).each do |key, value|
+                  - if value
+                    %li.custom-csv-error
+                      = link_to(value,
+                                "##{key}-error",
+                                id: "#{key}-error")
```
Comment (likeuwill) [inline app/views/users_management/users/set_up.html.haml:29]: one line

File: app/views/users_management/users/set_up.html.haml [hunk 3]
```diff
@@ -1,8 +1,62 @@
...
+
+      %p
+        You have been invited to set up a Clean Air Zone account.
+      %p
+        Create a password so that you can sign in.
+
+      = form_for('account_set_up', url: confirm_set_up_users_path(token: @token), method: :post) do |f|
+        = f.hidden_field :token, :value => @token
```
Comment (likeuwill) [inline app/views/users_management/users/set_up.html.haml:37]: `
value: @token
`

File: app/forms/set_up_account_form.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  validates :password,
+            presence: { message: I18n.t('email_and_password_form.password_missing') }
+
+  validates :password_confirmation,
+            presence: { message: I18n.t('email_and_password_form.password_confirmation_missing') }
+
+  validate :correct_password_confirmation?
```
[the remark this answers is not in the document]
Comment (pkalamarski) [reply app/forms/set_up_account_form.rb]: `validates :password_confirmation, presence: { message: I18n.t('email_and_password_form.password_confirmation_missing') }` is too long for rubocop, left it as it were

**Pull request merged**